src/FolderConfig.cpp

```diff
@@ -0,0 +1,94 @@
+#include "FolderConfig.h"
+#include <QFile>
+#include <QDebug>
+#include<QDir>
+
+namespace QtDBMigrationNS {
+FolderConfig::FolderConfig(const QString &path): AbstractConfig(path)
+{
+}
+
+bool FolderConfig::load()
+{
+    QDir dir(m_path);
+    QStringList nameFilter("*.sql");
+    QStringList sqlFiles = dir.entryList(nameFilter);
+    int totalFiles = sqlFiles.count();
+
+    if (!dir.exists() || totalFiles < 1) {
+        return false;
+    }
+
+    for(auto filename : sqlFiles) {
+        bool migrationsUp = true;
+        QString filePath = dir.filePath(filename);
+        QFile file(filePath);
+        if (!file.exists()) {
+            return false;
+        }
+
+        QFileInfo fileInfo(file.fileName());
+        QRegExp rx("^\\d{4}\\.sql$");
```

Comment:
Ah, I see you have an exact match here. Use the number out of here. No need for second regex.

Reply:
Changed to use number from the first RegEx.